python/foglamp/plugins/south/sinusoid/sinusoid.py

```diff
@@ -166,35 +168,48 @@ def generate_data():
 
     async def save_data():
         try:
-            while True:
-                time_stamp = utils.local_timestamp()
-                data = {
-                    'asset': handle['assetName']['value'],
-                    'timestamp': time_stamp,
-                    'key': str(uuid.uuid4()),
-                    'readings': {
-                        "sinusoid": next(generate_data())
-                    }
+            time_stamp = utils.local_timestamp()
+            data = {
+                'asset': handle['assetName']['value'],
+                'timestamp': time_stamp,
+                'key': str(uuid.uuid4()),
+                'readings': {
+                    "sinusoid": next(generate_data())
                 }
-
-                await Ingest.add_readings(asset='{}'.format(data['asset']),
-                                          timestamp=data['timestamp'], key=data['key'],
-                                          readings=data['readings'])
-
-                try:
-                    await asyncio.sleep(1 / (float(handle['dataPointsPerSec']['value'])))
-                except ZeroDivisionError:
-                    _LOGGER.warning('Data points per second must be greater than 0, defaulting to 1')
-                    handle['dataPointsPerSec']['value'] = '1'
-                    await asyncio.sleep(1)
-        except asyncio.CancelledError:
+            }
+            await Ingest.add_readings(asset='{}'.format(data['asset']),
+                                      timestamp=data['timestamp'], key=data['key'],
+                                      readings=data['readings'])
+        except (RuntimeWarning, asyncio.CancelledError):
             pass
         except (Exception, RuntimeError) as ex:
             _LOGGER.exception("Sinusoid exception: {}".format(str(ex)))
             raise exceptions.DataRetrievalError(ex)
 
-    _task = asyncio.ensure_future(save_data())
-
+    def run_task(loop):
+        global _task, should_stop
+        # _LOGGER.warning('>>>>>>> No. of active threads: %d, time: %s', active_count(), time.time())
```

Comment:
Remove it, or comment with `_LOGGER.info('No. of active threads: %d, time: %s', active_count(), time.time())`